src/SimpleAmqpClient/Channel.h

```diff
@@ -89,9 +90,11 @@ class SIMPLEAMQPCLIENT_EXPORT Channel : boost::noncopyable
                         const std::string &username = "guest",
                         const std::string &password = "guest",
                         const std::string &vhost = "/",
-                        int frame_max = 131072)
+                        int frame_max = 131072,
+                        const amqp_sasl_method_enum sasl_method=AMQP_SASL_METHOD_PLAIN)
```

Comment:
The call for this function when sasl_method is something other than PLAIN now becomes strange. What should username/password be set to when AMQP_SASL_METHOD_EXTERNAL? How about a SASL method that takes a different parameter all together.

It probably needs to be a completely different constructor.